### PR TITLE
[modules/kernel] Implement sceKernelGetModuleInfoByAddr

### PR DESCRIPTION
**Pull request by frangarcj**

### CODE REVIEW

**Comment by Zangetsu38** [comment]
Broken ingame/menu on jetpack royride and maybe other so

File: src/emulator/modules/SceLibKernel/SceLibKernel.cpp [hunk 1]
```diff
@@ -978,8 +978,21 @@ EXPORT(int, sceKernelGetLwMutexInfoById) {
...
+            const auto segment_address_begin = module->second->segments[n].vaddr.address();
+            const auto segment_address_end = segment_address_begin+module->second->segments[n].memsz;
+            if (addr.address() >segment_address_begin && addr.address() < segment_address_end) {
+                memcpy(info, module->second.get(), sizeof(emu::SceKernelModuleInfo));
+                return SCE_KERNEL_OK;
+            }
+        }
+    }
+
+    return SCE_KERNEL_ERROR_MODULEMGR_NOENT;
```
Comment (Rinnegatamante) [inline src/emulator/modules/SceLibKernel/SceLibKernel.cpp]: RET_ERROR missing

File: src/emulator/modules/SceLibKernel/SceLibKernel.cpp [hunk 2]
```diff
@@ -978,8 +978,21 @@ EXPORT(int, sceKernelGetLwMutexInfoById) {
     return UNIMPLEMENTED();
 }
 
-EXPORT(int, sceKernelGetModuleInfoByAddr) {
-    return UNIMPLEMENTED();
+EXPORT(int, sceKernelGetModuleInfoByAddr, Ptr<void> addr, emu::SceKernelModuleInfo *info) {
+    KernelState *const state = &host.kernel;
+
+    for (SceKernelModuleInfoPtrs::iterator module = state->loaded_modules.begin(); module != state->loaded_modules.end(); ++module){
```
Comment (VelocityRa) [inline src/emulator/modules/SceLibKernel/SceLibKernel.cpp]:
```suggestion
    for (const auto &module : state->loaded_modules) {
```

File: src/emulator/modules/SceLibKernel/SceLibKernel.cpp [hunk 3]
```diff
@@ -978,8 +978,21 @@ EXPORT(int, sceKernelGetLwMutexInfoById) {
...
+        for(int n = 0; n < MODULE_INFO_NUM_SEGMENTS; n++) {
```
Comment (VelocityRa) [inline src/emulator/modules/SceLibKernel/SceLibKernel.cpp]:
```suggestion
        for (int n = 0; n < MODULE_INFO_NUM_SEGMENTS; n++) {
```

File: src/emulator/modules/SceLibKernel/SceLibKernel.cpp [hunk 4]
```diff
@@ -978,8 +978,21 @@ EXPORT(int, sceKernelGetLwMutexInfoById) {
...
+            const auto segment_address_begin = module->second->segments[n].vaddr.address();
+            const auto segment_address_end = segment_address_begin+module->second->segments[n].memsz;
```
Comment (VelocityRa) [inline src/emulator/modules/SceLibKernel/SceLibKernel.cpp]:
```suggestion
            const auto segment_address_end = segment_address_begin + module->second->segments[n].memsz;
```

File: src/emulator/modules/SceLibKernel/SceLibKernel.cpp [hunk 5]
```diff
@@ -978,8 +978,21 @@ EXPORT(int, sceKernelGetLwMutexInfoById) {
...
+            if (addr.address() >segment_address_begin && addr.address() < segment_address_end) {
```
Comment (VelocityRa) [inline src/emulator/modules/SceLibKernel/SceLibKernel.cpp]:
```suggestion
            if (addr.address() > segment_address_begin && addr.address() < segment_address_end) {
```

**Pull request merged**